### PR TITLE
Add `Flux.gpu` function for `TemporalSnapshotsGNNGraph` type

### PR DESCRIPTION
**Pull request by aurorarossi**

### CODE REVIEW

File: src/GNNGraphs/temporalsnapshotsgnngraph.jl [hunk 1]
```diff
@@ -239,4 +239,14 @@ function print_feature_t(io::IO, feature)
...
+
+# move temporal graph to gpu
+function Flux.gpu(tsg::TemporalSnapshotsGNNGraph)
+    return TemporalSnapshotsGNNGraph(
+        Flux.gpu(tsg.num_nodes),
+        Flux.gpu(tsg.num_edges),
+        tsg.num_snapshots,
+        Flux.gpu(tsg.snapshots),
+        Flux.gpu(tsg.tgdata))
 end
```
Comment (CarloLucibello) [inline src/GNNGraphs/temporalsnapshotsgnngraph.jl:252]: I think we can avoid implementing gpu by just defining `@functor TemporalSnapshotsGNNGraph`

File: test/GNNGraphs/temporalsnapshotsgnngraph.jl [hunk 2]
```diff
@@ -101,3 +101,16 @@ end
     tsg.tgdata.x=rand(4)
     @test sprint(show,tsg) == "TemporalSnapshotsGNNGraph(5) with x: 4-element data"
 end
+
+@testset "gpu" begin
```
Comment (CarloLucibello) [inline test/GNNGraphs/temporalsnapshotsgnngraph.jl:105]:
```suggestion
if TEST_GPU
    @testset "gpu" begin
```

**Comment by CarloLucibello** [comment]
Did you run the gpu test locally to see if it passes?

**Comment by aurorarossi** [comment]
Yes!

**Pull request merged**